src/lib/metrics/regression.ts

```diff
@@ -28,29 +28,17 @@ export function mean_absolute_error(
     sample_weight: null,
   },
 ): number {
-  const yTrueShape = inferShape(y_true);
-  const yPredShape = inferShape(y_pred);
-
   // Validation 1: empty array check
-  if (yTrueShape[0] === 0 || yPredShape[0] === 0) {
-    throw new TypeError(`y_true ${JSON.stringify(y_true)} and y_pred ${JSON.stringify(y_pred)} cannot be empty`);
-  }
+  const [yTrueTensor, yPredTensor]: tf.Tensor[] = compareShapesAndConvert(y_true, y_pred);
```

Comment:
Can we put `compareShapesAndConvert` into `utils/validation.ts` for consistency?

Also, let's keep the naming consistent to what we already have

I think we can name it like `validateShapesEqual` and the function must return an array of tensors as a result.

Reply:
Done